modules/miner/blockmanager_test.go

```diff
@@ -0,0 +1,105 @@
+package miner
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/NebulousLabs/Sia/types"
+)
+
+// reconstructBlock reconstructs a block given the miner and the header
+func reconstructBlock(m *Miner, header types.BlockHeader) (block *types.Block, err error) {
+	block, exists := m.blockMem[header]
+	if !exists {
+		err = errors.New("Header is either invalid or too old")
+		return
+	}
+	arbData := m.arbDataMem[header]
+
+	block.Transactions[0].ArbitraryData[0] = arbData
+	return
+}
+
+// mineHeader takes a header, and nonce grinds it. It returns
+// the header with a nonce that solves the corresponding block
+func mineHeader(m *Miner, header types.BlockHeader) (bh types.BlockHeader, err error) {
+	b, err := reconstructBlock(m, header)
+	if err != nil {
+		return
+	}
+	*b, _ = m.SolveBlock(*b, m.target)
+	bh = b.Header()
+	return
+}
+
+// TestBlockManager creates a miner, then polls the Miner for block
+// headers to mine. It ensures that the blockmanager properly hands
+// out headers, then reconstructs the blocks
+func TestBlockManager(t *testing.T) {
+	mt, err := createMinerTester("TestBlockManager")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	headers := make([]types.BlockHeader, 2*headerForWorkMemory)
```

Comment:
you should probably write a comment indicating the reason it's 2*headerForWorkMemory